File: backend/app/schemas/organizer.py
from uuid import UUID
from datetime import date

from pydantic import BaseModel


class OrganizerHackathonResponse(BaseModel):
    id: UUID
    title: str
    description: str
    organizer: str
    mode: str
    location: str
    team_size: int
    difficulty: str
    prize_pool: str
    registration_deadline: str
    start_date: str
    end_date: str
    banner_image: str
    website: str
    status: str

    model_config = {
        "from_attributes": True
    }


class OrganizerStatsResponse(BaseModel):
    hackathons: int
    participants: int
    teams: int
    projects: int


class OrganizerHackathonUpdate(BaseModel):
    title: str | None = None
    description: str | None = None
    mode: str | None = None
    location: str | None = None
    team_size: int | None = None
    difficulty: str | None = None
    prize_pool: str | None = None
    registration_deadline: date | None = None
    start_date: date | None = None
    end_date: date | None = None
    banner_image: str | None = None
    website: str | None = None

File: frontend/services/organizer.ts
import { api } from "@/lib/api";

// ============================================================
// ORGANIZER DASHBOARD
// ============================================================

export async function getOrganizerStats() {
  const response = await api.get("/organizer/stats");
  return response.data;
}

// ============================================================
// ORGANIZER HACKATHONS
// ============================================================

export async function getOrganizerHackathons() {
  const response = await api.get("/organizer/hackathons");
  return response.data;
}

// ============================================================
// CREATE HACKATHON
// ============================================================

export async function createOrganizerHackathon(data: {
  title: string;
  description: string;
  organizer?: string;
  mode: string;
  location: string;
  team_size: number;
  difficulty: string;
  prize_pool: string;
  registration_deadline: string;
  start_date: string;
  end_date: string;
  banner_image?: string;
  website?: string;
}) {
  const response = await api.post(
    "/hackathons/",
    {
      ...data,
      organizer: data.organizer || "",
    }
  );

  return response.data;
}

// ============================================================
// GET SINGLE HACKATHON
// ============================================================

export async function getOrganizerHackathon(
  id: string
) {
  const response = await api.get(
    `/hackathons/${id}`
  );

  return response.data;
}

// ============================================================
// UPDATE HACKATHON
// ============================================================

export async function updateOrganizerHackathon(
  id: string,
  data: Record<string, unknown>
) {
  const response = await api.put(
    `/organizer/hackathons/${id}`,
    data
  );

  return response.data;
}

// ============================================================
// OPEN / CLOSE HACKATHON
// ============================================================

export async function updateOrganizerHackathonStatus(
  id: string,
  status?: string
) {
  const response = await api.patch(
    `/organizer/hackathons/${id}/status`
  );

  return response.data;
}

// ============================================================
// DELETE HACKATHON
// ============================================================

export async function cancelOrganizerHackathon(
  id: string
) {
  const response = await api.delete(
    `/organizer/hackathons/${id}`
  );

  return response.data;
}